rig/place_and_route/utils.py

```diff
@@ -299,8 +294,54 @@ def build_routing_tables(routes, net_keys, omit_default_routes=True):
     -------
     {(x, y): [:py:class:`~rig.routing_table.RoutingTableEntry`, ...]
     """
+    # Build full routing tables and then remove default entries from them
+    tables = dict()
+
+    for chip, table in iteritems(build_full_routing_tables(routes, net_keys)):
+        if omit_default_routes:
+            table = remove_default_routes(table, target_length=None)
+
+        # If the table is empty don't add it to the dictionary of tables.
+        if table:
+            tables[chip] = table
+
+    return tables
+
+
+def build_full_routing_tables(routes, net_keys):
```

Comment:
I don't like this name. :( Alas, the ideal name is still probably build_routing_tables... Ugh... Any other ideas(?) I'm stuck :(.